### PR TITLE
serialize global curried objects.

### PR DESCRIPTION
**Pull request by eriknw**
Use custom serialization logic via `curry.__reduce__`.  Issue #324.
    
This allows module-level globals (and their derivatives) created via `@curry`  to be serialized.  For example, `toolz.functoolz.flip` is curried, and the  original curried function is not resolvable by an import path.

Instead of returning a state via `curry.__getstate__` that includes the func  that is no longer resolvable, we return the name of the function if it is  the original (i.e., `toolz.functoolz.flip` will return `"flip"`).  For derived instances such as `flip(f)` (which creates a new curried instance), we return a resolvable import path such as `"toolz.functoolz.flip"` instead of the function, because the function can't be pickled.
    
More within-code documentation needed.

Remove Python 2.6 from test matrix so we can use `importlib`

This is a bit fancy, and there's a pretty decent chance this may introduce new corner cases.

### CODE REVIEW

File: .binstar.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ platform:
   - win-64
   - win-32
 engine:
-  - python=2.6
```
Comment (mrocklin) [inline .binstar.yml]: This is a biggish change, no?

Comment (eriknw) [inline .binstar.yml]: I guess so.  It would be straightforward to support 2.6.  It was easier to drop it.  What's your preference?

Comment (mrocklin) [inline .binstar.yml]: Given that I'm not actively doing much of the work on Toolz these days I'll
defer to your opinion.

However, I think that if we do drop Python 2.6 then we should first issue a
call out to the general community and have people weigh in.  Perhaps raise
an issue and we tweet about it?

Comment (eriknw) [inline .binstar.yml]: Fair 'nuff.  Re-added Python 2.6.  It wasn't that painful--this time.

Comment (mrocklin) [inline .binstar.yml]: I'm also entirely happy to drop Python 2.6 if you think we've reached that
point.  My only request is that we ask for input for a suitable period
before dropping it.

**Comment by llllllllll** [comment]
Thanks for this fix!
